modules/reactor/src/meshgenerators/PolygonConcentricCircleMeshGeneratorBase.C

```diff
@@ -881,6 +890,97 @@ PolygonConcentricCircleMeshGeneratorBase::generate()
           i + interface_id_shift + _interface_boundary_id_shift) = _interface_boundary_names[i];
     }
   }
+
+  if (isParamValid("sector_id_name"))
+  {
+    const auto extra_id_index = mesh0->add_elem_integer(getParam<std::string>("sector_id_name"));
+    // vector to store sector ids for each element
+    std::vector<dof_id_type> elem_sector_ids(mesh0->n_elem(), 0);
+    unsigned int id = 1;
+    // starting element id of the current sector
+    unsigned int offset = 0;
+    for (unsigned int is = 0; is < _num_sides; ++is)
+    {
+      // number of elements in the current sector
+      unsigned int nelem_sector =
+          mesh0->n_elem() * _num_sectors_per_side[is] /
+          (accumulate(_num_sectors_per_side.begin(), _num_sectors_per_side.end(), 0));
+      // set sector ids in the current sector
+      std::fill(
+          elem_sector_ids.begin() + offset, elem_sector_ids.begin() + offset + nelem_sector, id);
+      // update starting element id of the next sector
+      offset += nelem_sector;
+      ++id;
+    }
+    // assign sector ids to mesh
+    unsigned int i = 0;
+    for (auto & elem : mesh0->element_ptr_range())
+      elem->set_extra_integer(extra_id_index, elem_sector_ids[i++]);
+  }
+
+  if (isParamValid("ring_id_name"))
+  {
+    const auto extra_id_index = mesh0->add_elem_integer(getParam<std::string>("ring_id_name"));
+    // vector to ring sector ids for each element
+    std::vector<dof_id_type> elem_ring_ids(mesh0->n_elem(), 0);
+    unsigned int offset_side = 0;
+    for (unsigned int is = 0; is < _num_sides; ++is)
+    {
+      // number of elements in the current sector
+      unsigned int nelem =
+          mesh0->n_elem() * _num_sectors_per_side[is] /
+          (accumulate(_num_sectors_per_side.begin(), _num_sectors_per_side.end(), 0));
+      unsigned int offset_ring = 0;
+      if (getParam<MooseEnum>("ring_id_assign_type") == "block_wise")
+      {
+        for (unsigned int ir = 0; ir < _ring_radii.size(); ++ir)
+        {
+          // number of elements in the current ring and sector
+          unsigned int nelem_annular_ring = _num_sectors_per_side[is] * _ring_intervals[ir];
+          // if _quad_center_elements is true, the number of elements in center ring are
+          // _num_sectors_per_side[is] * _num_sectors_per_side[is] / 4
+          if (_quad_center_elements && ir == 0)
+            nelem_annular_ring = _num_sectors_per_side[is] * (_ring_intervals[ir] - 1) +
+                                 _num_sectors_per_side[is] * _num_sectors_per_side[is] / 4;
+          // set ring ids in the current ring block
+          std::fill(elem_ring_ids.begin() + offset_side + offset_ring,
+                    elem_ring_ids.begin() + offset_side + offset_ring + nelem_annular_ring,
+                    ir + 1);
```

Comment:
this one would be awkward though a double-strided assignment 
maybe not a good idea then